From 963fbe9b26cecd945eff31b49b2813928eaa1404 Mon Sep 17 00:00:00 2001
From: Oliver Heger <oheger@apache.org>
Date: Sat, 6 Mar 2010 20:39:11 +0000
Subject: [PATCH] CONFIGURATION-409: Added a test for INIConfigurationSource to
 ensure that this class is not affected by this issue.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/configuration/branches/configuration2_experimental@919845 13f79535-47bb-0310-9956-ffa450edef68
---
 .../base/impl/TestINIConfigurationSource.java | 25 +++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/src/test/java/org/apache/commons/configuration2/base/impl/TestINIConfigurationSource.java b/src/test/java/org/apache/commons/configuration2/base/impl/TestINIConfigurationSource.java
index 0c35eb21b0..eabf9158f8 100644
--- a/src/test/java/org/apache/commons/configuration2/base/impl/TestINIConfigurationSource.java
+++ b/src/test/java/org/apache/commons/configuration2/base/impl/TestINIConfigurationSource.java
@@ -656,4 +656,29 @@ public void testSaveToLocator() throws IOException, ConfigurationException
                 source2);
         checkContent(config);
     }
+
+    /**
+     * Tests whether section keys with delimiters in their names cause problems.
+     * This test is related to CONFIGURATION-409.
+     */
+    @Test
+    public void testSaveKeysWithDelimiters() throws IOException,
+            ConfigurationException
+    {
+        Configuration<ConfigurationNode> conf = new ConfigurationImpl<ConfigurationNode>(
+                source);
+        final String section = "Section..with..dots";
+        conf.addProperty(section + ".test1", "test1");
+        conf.addProperty(section + ".test2", "test2");
+        StringWriter writer = new StringWriter();
+        source.save(writer);
+        StringReader reader = new StringReader(writer.toString());
+        source = new INIConfigurationSource();
+        source.load(reader);
+        conf = new ConfigurationImpl<ConfigurationNode>(source);
+        assertEquals("Wrong value (1)", "test1", conf.getString(section
+                + ".test1"));
+        assertEquals("Wrong value (2)", "test2", conf.getString(section
+                + ".test2"));
+    }
 }
